src/plugins/dashboard/public/application/top_nav/dashboard_top_nav.tsx

```diff
@@ -262,44 +260,66 @@ export function DashboardTopNav({
         confirmOverwrite: false,
         isTitleDuplicateConfirmed,
         onTitleDuplicate,
+        saveAsCopy: newCopyOnSave,
       };
       const stateFromSaveModal = {
         title: newTitle,
         description: newDescription,
         timeRestore: newTimeRestore,
         tags: [] as string[],
       };
-      if (hasApi && newTags) {
-        // remove `hasAPI` once the savedObjectsTagging service is optional
+      if (hasSavedObjectsTagging && newTags) {
+        // remove `hasSavedObjectsTagging` once the savedObjectsTagging service is optional
         stateFromSaveModal.tags = newTags;
       }
 
-      dashboardAppState.savedDashboard.copyOnSave = newCopyOnSave;
-      const saveResult = await saveDashboard({
+      /**
+       * Check for duplicate title
+       */
+      try {
+        await checkForDuplicateDashboardTitle({
+          title: newTitle,
+          onTitleDuplicate,
+          lastSavedTitle: currentState.title,
+          copyOnSave: newCopyOnSave,
+          isTitleDuplicateConfirmed,
+        });
+      } catch (error) {
+        if (
+          error &&
+          dashboardSavedObjectErrorStrings.getSaveDuplicateTitleRejected() === error.message
+        ) {
+          return { id: '' };
+        }
+        return { error };
+      }
```

Comment:
Could we make this a function so that the title duplication check isn't more-or-less duplicated in both `runSaveAs` and `runClone`? Would probably make this a bit cleaner 👍

Reply:
Whoops, meant to do this before opening the PR - will figure out a more DRY implementation of this.